File: app/src/main/java/org/michaelbel/tjgram/data/entity/Cover.java
package org.michaelbel.tjgram.data.entity;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Cover implements Serializable {
    public static final int TYPE_IMAGE = 1;
    public static final int TYPE_VIDEO = 2;

    @Expose @SerializedName("additionalData") public AdditionalData additionalData;
    @Expose @SerializedName("size") public Size size;
    @Expose @SerializedName("size_simple") public String sizeSimple;
    @Expose @SerializedName("thumbnailUrl") public String thumbnailUrl;
    @Expose @SerializedName("type") public Integer type;
    @Expose @SerializedName("url") public String url;
}

File: app/src/main/java/org/michaelbel/tjgram/data/entities/Cover.kt
package org.michaelbel.tjgram.data.entities

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.io.Serializable

data class Cover (
    @Expose @SerializedName("additionalData") var additionalData: AdditionalData,
    @Expose @SerializedName("size") var size: Size,
    @Expose @SerializedName("size_simple") var sizeSimple: String,
    @Expose @SerializedName("thumbnailUrl") var thumbnailUrl: String,
    @Expose @SerializedName("type") var type: Int,
    @Expose @SerializedName("url") var url: String
): Serializable {

    fun isImage(): Boolean = type == 1

    fun isVideo(): Boolean = type == 2
}